Amazon.QLDB.Driver/executable/IAsyncExecutable.cs

```diff
@@ -0,0 +1,57 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.QLDB.Driver
+{
+    using System.Collections.Generic;
+    using Amazon.IonDotnet.Tree;
+    using Amazon.Runtime;
+
+    /// <summary>
+    /// Interface for executions of a statement within an active transaction to QLDB.
+    /// </summary>
+    public interface IAsyncExecutable
+    {
+        /// <summary>
+        /// Execute the statement against QLDB and retrieve the result.
+        /// </summary>
+        ///
+        /// <param name="statement">The PartiQL statement to be executed against QLDB.</param>
+        ///
+        /// <returns>Result from executed statement.</returns>
+        /// <exception cref="AmazonServiceException">Thrown when there is an error executing against QLDB.</exception>
+        IAsyncResult Execute(string statement);
```

Comment:
The signatures for these are all incorrect.

Reply:
These are not intended to be the final signatures.